### PR TITLE
25961 doc structure

### PR DESCRIPTION
**Pull request by md-magenta**
I have made a new general structure of the documentation. There a very few addition. Its mostly the old text moved around and duplicated text removed. 

Supersedes #96

### CODE REVIEW

File: doc/dev/test.rst [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+Make sure the parameter ``USE_SAML_AUTHENTICATION`` in
+``oio_rest/oio_rest/settings.py`` is `False`.
+
+Make sure the parameter ``moxrestfrontend.rest.host`` in
+``agents/MoxRestFrontend/moxrestfrontend.conf`` is set to
+``http://localhost:5000``.
+
+Start the (AMQP) MOX REST frontend agent::
+
```
Comment (agger-magenta) [inline doc/dev/test.rst]: This doesn't work right now, does it? Maybe we should omit this and the thing about `test.sh`.

File: doc/dev/test.rst [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+This should give you a lot of output like this::
+
+    Deleting bruger, uuid: 1e874f85-07e5-40e5-81ed-42f21fc3fc9e
+    Getting authtoken
+    127.0.0.1 - - [27/Apr/2016 15:55:09] "DELETE /organisation/bruger/1e874f85-07e5-40e5-81ed-42f21fc3fc9e HTTP/1.1" 200 -
+    Delete succeeded
+
+**Note:** Currently, some of the tests will give the notice: "Result differs
+from the expected". This is due to a bug in the tests, i.e. you should not
+worry about this — if you see output as described above, the system is working.
```
Comment (agger-magenta) [inline doc/dev/test.rst]: That is, I propose to delete up to and including this note.

File: doc/dev/test.rst [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+The install.sh script creates an Apache VirtualHost for oio rest and
+MoxDocumentUpload.
+
+To run the OIO Rest Mox Agent (the one listening for messages and
+relaying them onwards to the REST interface), run::
+
+    $ agents/MoxRestFrontend/moxrestfrontend.sh
+
+NOTE:** You can start the agent in the background by running::
```
Comment (agger-magenta) [inline doc/dev/test.rst]: This is old-style, please use `.. note::` for notes in RST.

File: doc/user/auth.rst [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+an error message will be returned.
+
+This value can then be included in the HTTP "Authorization" header, like the
+following::
+
+    Authorization: <output of get-token>
+
+For testing purposes, we recommend the browser extensions `Advanced
+REST client`_ for Chrome or `REST Easy`_ for Firefox.
+
```
Comment (agger-magenta) [inline doc/user/auth.rst]: or `curl`.

File: doc/user/overview.rst [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    Ideally used with a frontend HTTP proxy in production.
+
+RabbitMQ
+    AMQP message broker providing interprocess communication between
+    the various components.
+
+
+Agents
+------
+
```
Comment (agger-magenta) [inline doc/user/overview.rst]: There's also `Mox Advis` and `Mox ELK Log` in the `python_agents` folder.

But I think all documentation about agents should have a notice that this is out of date, as we don't currently maintain any of the agents.

**Comment by agger-magenta** [comment]
Bemærk, at vores konvention for feature-branches er, at de begynder med `feature`: Altså 

`feature/<ticket_number>_some_description`

Denne konvention kommer fra Git Flow, som vi bruger en variant af.

**Comment by md-magenta** [comment]
Jenkins fejlede. Jeg har rebased på `development` og det virker nu. @agger-magenta vil du approve igen?

**Pull request merged**